src/environments/suweight.jl

```diff
@@ -125,6 +125,25 @@ function SUWeight(peps::InfinitePEPS)
     return SUWeight(Nspaces, Espaces)
 end
 
+"""
+    SUWeight(pepo::InfinitePEPO)
+
+Create an SUWeight for a given one-layer InfinitePEPO.
+The weights are initialized as identity matrices of element type `Float64`.
+"""
+function SUWeight(pepo::InfinitePEPO)
+    @assert size(pepo, 3) == 1
+    Nspaces = map(pepo.A[:, :, 1]) do t
```

Comment:
```suggestion
    Nspaces = map(@view(pepo.A[:, :, 1])) do t
```

Here you have to be a little bit careful because slicing arrays, even in this trivial case, allocates a new array, which isn't necessary. This is not a huge impact, but might as well avoid it here.